### PR TITLE
[CodeQuality] Skip intersection property var type on ImproveDoctrineCollectionDocTypeInEntityRector

### PR DESCRIPTION
**Pull request by samsonasik**
Closes https://github.com/rectorphp/rector-doctrine/issues/342

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by bobvandevijver** [comment]
This would only skip the rule, which is fine by me, but note that I also tried to create #343 which would also insert the correct Selectable annotation.

File: rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Fixture/OneToMany/Attribute/skip_collection_intersection_selectable.php.inc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+use Rector\Doctrine\Tests\CodeQuality\Rector\Property\ImproveDoctrineCollectionDocTypeInEntityRector\Source\CountryRef;
+
+#[ORM\Entity]
+class SkipCollectionIntersectionSelectable
+{
+	/**
+	 * @var Collection<int, CountryRef>&Selectable
+	 */
+	#[ORM\OneToMany(mappedBy: 'country', targetEntity: CountryRef::class, cascade: ['persist'])]
+	private Collection $countryRefs;
```
Comment (TomasVotruba) [inline rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Fixture/OneToMany/Attribute/skip_collection_intersection_selectable.php.inc:18]: While this fixes proposed issue, it doesn't make sense from PHP points of view.

This code should be updated as well:

```php
private Collection $countryRefs;
```

Also, in the case of using `matching()` in many places, this would be correct:

```php
private ArrayCollection $countryRefs;
```

But this is out of scope of this rule. I'd recommend to create custom rule to use `ArrayCollection` if possible.

Comment (bobvandevijver) [reply rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Fixture/OneToMany/Attribute/skip_collection_intersection_selectable.php.inc:18]: It is not possible to use `ArrayCollection` as type.  Depending on the state of the object, it is either `ArrayCollection` or `PersistentCollection`.
